File: app/src/main/java/com/wen/android/mtabuscomparison/util/SearchHandler.java
package com.wen.android.mtabuscomparison.util;

/**
 * Created by yuan on 4/21/2017.
 */

public class SearchHandler {
    private String mKeyword;

    public SearchHandler(String keyword){
        mKeyword = keyword;
    }

    public int keywordType(){
        String firstCharacter = Character.toString(mKeyword.charAt(0));
        //if return true then it is a stop code otherwise is a route
        boolean isStopCode = firstCharacter.matches("\\d+");
        if (isStopCode == true){
            //a stop code
            return 0;
        } else {
            // a route
            return 1;
        }
    }
}

File: app/src/main/java/com/wen/android/mtabuscomparison/util/SearchHandler.kt
package com.wen.android.mtabuscomparison.util

import com.wen.android.mtabuscomparison.feature.search.SearchType


class SearchHandler(private val mKeyword: String) {

    fun keywordType(): SearchType {
        val firstCharacter = Character.toString(mKeyword[0])
        //if return true then it is a stop code otherwise is a route
        val isStopCode = firstCharacter.matches(Regex("\\d+"))
        return if (isStopCode) {
            SearchType.STOP
        } else {
            SearchType.ROUTE
        }
    }

}
